tests/unit/s2n_server_new_session_ticket_test.c

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "tls/s2n_tls.h"
+/* To test static functions */
+#include "tls/s2n_server_new_session_ticket.c"
+
+#define ONE_HOUR_IN_NANOS   3600000000000
+#define TWO_HOURS_IN_NANOS  ONE_HOUR_IN_NANOS * 2
+#define TICKET_AGE_ADD_MARKER sizeof(uint8_t)  + /* message id  */ \
+                              SIZEOF_UINT24    + /* message len */ \
+                              sizeof(uint32_t)   /* ticket lifetime */
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+    
+    /* s2n_tls13_server_nst_send */
+    {
+        /* Session ticket keys. Taken from test vectors in https://tools.ietf.org/html/rfc5869 */
```

Comment:
You can reference the RFC with the tool which makes it easier to check that they're correct